src/tests/Microsoft.PowerFx.Core.Tests/IRTests/IRPrettyPrinter.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.IR.Symbols;
+
+namespace Microsoft.PowerFx.Tests
+{
+    // Pretty printer for tests. 
+    // Stable  - tests rely on this, so we need to fix the format 
+    // Compact / No newlines - ideally can fit on a single line in Theorys
+    // Deterministic ordering - when traversing dictionaries
+    // Unambiguous - additional parenthesis as needed to show (a+b)+c vs. a+(b+c). 
+    internal class PrettyPrintIRVisitor : IRNodeVisitor<PrettyPrintIRVisitor.RetVal, PrettyPrintIRVisitor.Context>
```

Comment:
Would be useful in Core, for CheckResult.PrintIR()

(which, also, should probably not be public on CheckResult??)

Reply:
starting in Test seemed safe. We could move in future. 
Note that this Compact form is much less invasive than the one in product.   More ideal for test baselines, less ideal for diagnostics.